backend/src/middleware/authMiddleware.js

```diff
@@ -0,0 +1,33 @@
+import jwt from "jsonwebtoken";
+import dotenv from "dotenv";
+dotenv.config();
+
+const accessKey = process.env.ACCESS_TOKEN_SECRET;
+
+const verifyToken = async (req, res, next) => {
+  try {
+    const token = req.cookies.jwt;
+
+    if (!token) return res.sendStatus(401);
+
+    jwt.verify(token, accessKey, (err, user) => {
+      if (err) return res.sendStatus(403);
+      req.user = user;
+      next();
+    });
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+const isAdmin = async (req, res, next) => {
+  try {
+    const { role } = req.user;
+    if (role !== "Admin") return res.sendStatus(401);
```

Comment:
Here, it should be a 403 forbidden error because the user is authenticated but trying to access an area which he should not have access to.